WIP rerender error

beckermr opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * The recipe `license` should not include the word "License".

For **recipe**:

 * License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).

beckermr commented on .ci_support/migrations/cuda111_112.yaml:
```diff
@@ -0,0 +1,95 @@
+      - quay.io/condaforge/linux-anvil-cuda:11.0    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cuda:11.1    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cuda:11.2    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cos7-cuda:9.2     # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cos7-cuda:10.0    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cos7-cuda:10.1    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cos7-cuda:10.2    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
```
I added only the extra ordering keys here.

beckermr commented on recipe/conda_build_config.yaml:
```diff
@@ -19,3 +19,23 @@ cuda_compiler_version:
   - 10.1                       # [linux64 or win]
   - 10.2                       # [linux64 or win]
   - 11.0                       # [linux64 or win]
+
+cudnn:
+  - undefined
+  - 7                   # [linux64]
+  - 7                   # [linux64 or win]
+  - 7                   # [linux64 or win]
+  - 7                   # [linux64 or win]
+  - 8                   # [linux64 or win]
+
+cdt_name:  # [linux]
+  - cos6   # [linux64]
+  - cos7   # [linux and aarch64]
+  - cos7   # [linux and ppc64le]
+  - cos7   # [linux and armv7l]
+
+  - cos7   # [linux64]
+  - cos7   # [linux64]
+  - cos7   # [linux64]
+  - cos7   # [linux64]
+  - cos6   # [linux64]
```
I put the extra zipping keys here.

beckermr commented:
ok @leofang @isuruf I have tried the changes discussed on gitter here. I am stumped. 

I added only the ordering keys and fixed the zipping. The feedstock rerenders, but the result doesn't add 11.1 or 11.2

beckermr commented on .ci_support/linux_64_cuda_compiler_version10.2.yaml:
```diff
@@ -13,9 +13,10 @@ cuda_compiler:
 - '10.2'
 docker_image:
-- quay.io/condaforge/linux-anvil-cos7-cuda:10.2
+- quay.io/condaforge/linux-anvil-cos7-cuda:9.2
```
Also this is wrong.

beckermr commented:
One tidbit here is that smithy is ignoring the ordering key in the migrator. :/

beckermr commented:
That key is only used for special key_add or key_remove operations in the migrators. cuda111_112.yaml is not one of those.

beckermr commented:
I think there is a bug in smithy where the CBC from the feedstock is applied twice. 

I also generalize key adds to work for more than one set of zip keys.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I was trying to look for recipes to lint for you, but it appears we have a merge conflict.
Please try to merge or rebase with the base branch to resolve this conflict.

Please ping the 'conda-forge/core' team (using the @ notation in a comment) if you believe this is a bug.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * The recipe `license` should not include the word "License".

For **recipe**:

 * License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

leofang commented:
Just FYI, starting cuDNN 8.1.0 for all CUDA 11.x there is a single source release. I haven't examined the content in the new package, but I suspect similar to cuTENSOR it also utilizes the binary compatibility feature on 11.x, so perhaps it's better to follow what I've done in cutensor-feedstock and release 1 conda package for all CUDA 11.x, saving us some headache here. The price to pay is to maintain a conda-forge-config.yml and reject migrators.

beckermr commented:
@conda-forge-admin rerender

beckermr commented:
New smithy is working fine here. I am going to close the PR but leave my branch. If people want these changes, we can put them into the pinnings repo and a new PR on cudnn
